plugins/peermanager/main.go

```diff
@@ -0,0 +1,108 @@
+package peermanager
+
+import (
+	"fmt"
+
+	"github.com/Shopify/sarama"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/openrelayxyz/xplugeth"
+	"github.com/openrelayxyz/xplugeth/hooks/initialize"
+	"github.com/openrelayxyz/xplugeth/types"
+)
+
+type peerManagerConfig struct {
+	BrokerURL string `yaml:"peerManager.broker.url"`
+}
+
+var cfg *peerManagerConfig
+
+var (
+	httpApiFlagName    = "http.api"
+	sessionStack       node.Node
+	sessionBrokers     []string
+	sessionKafkaConfig *sarama.Config
+	nodes              = make(chan string, 5)
+	exit               = make(chan struct{}, 1)
+	peerBroker         string
+)
+
+type peerManagerModule struct{}
+
+func init() {
+	xplugeth.RegisterModule[peerManagerModule]("peerManagerModule")
+}
+
+func (*peerManagerModule) InitializeNode(s *node.Node, b types.Backend) {
+	sessionStack = *s
+	var ok bool
+	cfg, ok = xplugeth.GetConfig[peerManagerConfig]("peerManager")
+	peerBroker = cfg.BrokerURL
+	if !ok {
+		log.Warn("did not acqire config, example plugin, all values set to default")
+	}
+	log.Info("Initialized node, peer manager plugin")
+}
+
+func (*peerManagerModule) Blockchain() {
+	if peerBroker == "" {
+		panic(fmt.Sprintf("no broker provided for peer manager plugin"))
+	}
+	go peeringSequence()
+}
+
+func peeringSequence() {
+	sessionPeerService, err := getPeerManager()
+	if err != nil {
+		log.Error("session peer service unavailable, peer manager plugin", "err", err)
+		return
+	}
+
+	selfNode, err := sessionPeerService.getEnode()
+	if err != nil {
+		log.Error("error calling getEnode from sessionService, peer manager plugin", "err", err)
+	}
+
+	chainTopic, err := sessionPeerService.chainIdResolver()
```

Comment:
Let's pull the topic from the config. We can use chainIdResolver() if it's not explicitly set, but making this configurable would let us have separate clusters of nodes we don't force to peer with eachother.